invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""CSL based Schema for Invenio RDM Records."""
+
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing, post_dump
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class CreatorSchemaCSL(Schema):
+    """Creator/contributor common schema."""
+
+    family = fields.Str(attribute="person_or_org.name")
+
+    @post_dump(pass_many=False)
+    def capitalize_name_type(self, data, **kwargs):
+        """Capitalize type."""
+        if data.get("nameType"):
+            data["nameType"] = data["nameType"].capitalize()
+
+        return data
+
+
+class CSLJSONSchema(Schema):
+    """DataCite 4.3 Marshmallow Schema."""
+
+    publisher = fields.Str(attribute='metadata.publisher')
+    DOI = fields.Method('get_DOI')
+    language = fields.Method('get_language')
+    title = fields.Method('get_title')
+    issued = fields.Method('get_issued')
+    abstract = fields.Method('get_abstract')
+    author = fields.List(
+        fields.Nested(CreatorSchemaCSL), attribute='metadata.creators')
+    note = fields.Method('get_note')
+    version = SanitizedUnicode(attribute="metadata.version")
+    # PIDS-FIXME: What about versioning links and related ids
```

Comment:
This comment was not taken into account?

Reply:
It's for PIDS, I think it should be removed (before merge)